substrate/utils/frame/benchmarking-cli/src/storage/read.rs

```diff
@@ -44,12 +44,15 @@ impl StorageCmd {
 		let mut keys: Vec<_> = client.storage_keys(best_hash, None, None)?.collect();
 		let (mut rng, _) = new_rng(None);
 		keys.shuffle(&mut rng);
+		let number_of_keys = (keys.len() * self.params.db_fraction as usize) / 100;
```

Comment:
It should probably error if the result is zero.

Reply:
We could do something like:

```rust
let number_of_keys = if number_of_keys != 0 {number_of_keys} else {keys.len()};
```

If we have number_of_keys = 0, we use all the keys by default. Another option would be to set the minimum limit to 1. WDYT?

Reply:
But all keys could also be empty afaik the check makes sense either way. Just returning an error is no problem.

Reply:
If the `number_of_keys` is `0` that can mean either (or both) `keys.len()` or `self.params.db_fraction` was `0` , but `self.params.db_fraction` is restricted via clap arguments `1..=100` so it cannot be `0`, in which case `keys.iter().take(0)` is equivalent to `keys.iter()` thereby keeping the old behavior in tact. I believe there's no special case to be handled here unless we're worried someone/something overriding `db_fraction` to `0` in which case that should be the thing being checked here.

Reply:
What i am saying is that the old behaviour was probably also wrong.